Cleaning up code on Language Switcher

heckdavid opened the pull request.

heckdavid committed:
First commit. html file created with boilerplate. Empty css file. Empty js file.

heckdavid committed:
Added to body tag in html

heckdavid committed:
Linked javascript in html and added in more javascript

heckdavid committed:
Added editorconfig file

heckdavid committed:
Added some basic styling

heckdavid committed:
Changed languageSection to language-section

heckdavid committed:
Update index.html

Fixed comma spacing on line 8

TheJaredWilcurt commented on style.css:
```diff
@@ -0,0 +1,49 @@
+}
+
+.title {
+    color: white;
```
You are using `white` here and `#ffffff` below. Be consistent with your color declarations in your CSS.
```suggestion
    color: #FFF;
```

TheJaredWilcurt commented on style.css:
```diff
@@ -0,0 +1,49 @@
+}
+
+.description {
+    color: #ffffff;
```
```suggestion
    color: #FFF;
```

TheJaredWilcurt commented on style.css:
```diff
@@ -0,0 +1,49 @@
+    text-decoration: none;
+    padding: 0 15px;
+    font-weight: bold;
+    background: #feac32;
+    color: #0a0a23;
```
This varies a lot from team to team, I prefer caps, but as long as the codebase is consistent, that's what matters.
```suggestion
    background: #FEAC32;
    color: #0A0A23;
```

TheJaredWilcurt commented on style.css:
```diff
@@ -0,0 +1,49 @@
+}
+
+.language-section {
+    background: #ffffff;
```
```suggestion
    background: #FFF;
```

TheJaredWilcurt commented on style.css:
```diff
@@ -0,0 +1,49 @@
+    justify-content: center;
+    align-items: center;
+    min-height: 100vh;
+    background: #0a0a23;
```
```suggestion
    background: #0A0A23;
```

TheJaredWilcurt commented on style.css:
```diff
@@ -0,0 +1,49 @@
+    max-width: 600px;
+    height: 500px;
+    padding: 0px;
+    color: black;
```
```suggestion
    color: #000;
```

TheJaredWilcurt commented on index.js:
```diff
@@ -0,0 +1,31 @@
+        "title": "Statischen Text übersetzen.",
+        "description": "Statischen Text übersetzen."
+    }
+}
```
```suggestion
}

```

Missing a return, which means either you did not come back to this file and hit save after adding the editor config file. Or you did not install the editor config plugin in your editor.

heckdavid replied:
I have added the editor config file but I'm not for sure if I saved after adding that file. I do have the editor config plugin installed.

I tried to add an empty return which seems like it doesn't show up unless when in the editor.

TheJaredWilcurt commented on index.js:
```diff
@@ -0,0 +1,31 @@
+const langSec = document.querySelector(".language-section");
```
Abbreviation leads to confusion. It makes code less predictable and guessable.

`langSec` => `languageSelection`

When a variable contains an HTML Node, it is extremely common for developers to denote this with an `El` suffix, which is unfortunately an abbreviation, but it's one you need to learn. Stands for "element".

```suggestion
const languageSectionEl = document.querySelector('.language-section');
```

TheJaredWilcurt commented on index.js:
```diff
@@ -0,0 +1,31 @@
+const textDesc = document.querySelector(".description");
+
+buttons.forEach(button => {
+    button.addEventListener('click', () => {
```
Almost all JS codebases use 2 space indentation for `.js` files (~90+%). Same for HTML (~99+%). CSS is more often 4 spaces but it is not uncommon to see it in 2 spaces (~60% 4 spaces).
```suggestion
  buttonEl.addEventListener('click', () => {
```

TheJaredWilcurt commented on index.js:
```diff
@@ -0,0 +1,31 @@
+const langSec = document.querySelector(".language-section");
+const buttons = document.querySelectorAll(".button");
+const textTitle = document.querySelector(".title");
+const textDesc = document.querySelector(".description");
```
Don't abbreviate.

```suggestion
const textDescriptionEl = document.querySelector('.description');
```

TheJaredWilcurt commented on index.js:
```diff
@@ -0,0 +1,31 @@
+const langSec = document.querySelector(".language-section");
+const buttons = document.querySelectorAll(".button");
+const textTitle = document.querySelector(".title");
```
```suggestion
const buttonEls = document.querySelectorAll('.button');
const textTitleEl = document.querySelector('.title');
```

TheJaredWilcurt commented on index.js:
```diff
@@ -0,0 +1,31 @@
+const textTitle = document.querySelector(".title");
+const textDesc = document.querySelector(".description");
+
+buttons.forEach(button => {
```
If you are using arrow functions ALWAYS wrap the arguments in parenthesis.
```suggestion
buttonEls.forEach((buttonEl) => {
```

TheJaredWilcurt commented on index.js:
```diff
@@ -0,0 +1,31 @@
+
+buttons.forEach(button => {
+    button.addEventListener('click', () => {
+        langSec.querySelector(".active").classList.remove(".active");
```
Most JS codebases (~80%) use single quotes.
```suggestion
        languageSectionEl.querySelector('.active').classList.remove('.active');
```

TheJaredWilcurt commented on index.js:
```diff
@@ -0,0 +1,31 @@
+buttons.forEach(button => {
+    button.addEventListener('click', () => {
+        langSec.querySelector(".active").classList.remove(".active");
+        button.classList.add("active");
```
```suggestion
        buttonEl.classList.add('active');
```

TheJaredWilcurt commented on index.js:
```diff
@@ -0,0 +1,31 @@
+        langSec.querySelector(".active").classList.remove(".active");
+        button.classList.add("active");
+
+        const attr = button.getAttribute("data-language");
```
```suggestion
        const attribute = buttonEl.getAttribute('data-language');
```

TheJaredWilcurt commented on index.js:
```diff
@@ -0,0 +1,31 @@
+
+        const attr = button.getAttribute("data-language");
+
+        textTitle.textContent = data[attr].title;
+        textDesc.textContent = data[attr].description;
```
```suggestion
        textTitleEl.textContent = dictionary[attribute].title;
        textDescriptionEl.textContent = dictionary[attribute].description;
```

TheJaredWilcurt commented on index.js:
```diff
@@ -0,0 +1,31 @@
+    });
+});
+
+const data = {
```
`data` is an incredibly generic name. Whenever possible, always imbue your variable names with meaning that describes the data they hold.

```suggestion
const dictionary = {
```

TheJaredWilcurt commented on index.js:
```diff
@@ -0,0 +1,31 @@
+});
+
+const data = {
+    "english": {
+        "title": "Translating static text.",
+        "description": "Translating static text"
+    },
+    "french": {
+        "title": "Traduire du texte statique.",
+        "description": "Traduire du texte statique."
+    },
+    "german": {
+        "title": "Statischen Text übersetzen.",
+        "description": "Statischen Text übersetzen."
+    }
```
If you wanted to store your dictionary in a JSON file, or a series of JSON files (`en.json`, `de.json`, `fr.json`) then this double-quotes surrounding everything approach would be the right (only) way to do it.

And on a larger project, where all the copy on the site/app amounts to hundreds of KB or even MB's of data, then it would be much more preferred to break it up. Imagine going to a site and downloading a 1MB file for English, and then having to wait for 40 more MB to download of every other language, when you'll never switch to them. So it is more performant to break them up into language specific files, so they can be loaded asynchronously, on big projects.

But that's overkill for this project.

I think it's fine to keep this here in this file, but we should change it to be written more like what most devs would do. Switch to single-quotes, and remove quotes from the keys if they aren't needed.

```suggestion
  english: {
    title: 'Translating static text.',
    description: 'Translating static text'
  },
  french: {
    title: 'Traduire du texte statique.',
    description: 'Traduire du texte statique.'
  },
  german': {
    title': 'Statischen Text übersetzen.',
    description: 'Statischen Text übersetzen.'
  }
```

TheJaredWilcurt commented on index.html:
```diff
@@ -0,0 +1,35 @@
+<!-- Boilerplate 1-10-->
```
Not sure what this or the other comment are trying to communicate. Should these be removed?

TheJaredWilcurt commented on index.html:
```diff
@@ -0,0 +1,35 @@
+        </div>
+
+    </div>
+    <script src="index.js"></script>
```
Move this to the `<head>` section with a [defer](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#defer).
```suggestion
    <script defer src="index.js"></script>
```

heckdavid committed:
Update index.html

Removed unnecessary comments above boilerplate and body tag.

heckdavid committed:
Update index.html

Moved index.js to head section and added defer.

heckdavid committed:
Update index.js

Changed langSec to languageSectionEl.

heckdavid committed:
Update index.js

Added El abbreviation to buttonEls and textTitleEl on line 2 and 3

heckdavid committed:
Update index.js

Got rid of abbreviation for description and added El.

heckdavid committed:
Update index.js

Wrapped buttonEl in parentheses.

heckdavid committed:
Update index.js

Added two space indentation on line 7

heckdavid committed:
Update index.js

Switched from doubles quotes to single quotes on line 8.

heckdavid committed:
Update index.js

Switched from double quotes to single quotes on line 9.

heckdavid committed:
Update index.js

Got rid of abbreviation on attribute, added El onto button and changed doubles to single quotes on data-language.

heckdavid committed:
Update index.js

Changed data to dictionary to be more descriptive.

heckdavid committed:
Update index.js

Added El to textTitle, swapped data to dictionary and got rid of attribute abbreviation on line 13 and 14.

heckdavid committed:
Update index.js

Got rid of quotes around keys not being used. Also changed from doubles to single quotes.

heckdavid committed:
Update style.css

Switch to uppercase letters on line 10 background color.

heckdavid committed:
Update style.css

Changed line 18 from from black to #000.

heckdavid committed:
Update style.css

Changed line 23 from white to #FFF.

heckdavid committed:
Update style.css

Changed line 31 and 32 from lowercase to capitals.

heckdavid committed:
Update style.css

Changed line 38 from white to #FFF.

heckdavid committed:
Update style.css

Changed line 48 from lowercase to capital.

heckdavid committed:
Update index.js

Changed Line 1, 2, and 3 from double quotes to single quotes.

heckdavid committed:
Update index.js

Added El onto button on line 7 and modified line 8 to be languageSectionEl.

heckdavid committed:
Update index.js

Added empty return on line 32.

TheJaredWilcurt reviewed (approved):
Everything else looks good. Only thing to change is inconsistent indentation in the JS file. Generally HTML and JS file are 2 spaces (like 80-90% of code bases). CSS can be 2 or 4, more of a toss up. it's more often 4 but only by a little.

If you are coming today, I think we'll get you set up for linting.

heckdavid committed:
Merge branch 'main' into working-branch
